packages/flutter/test/widgets/semantics_tester.dart

```diff
@@ -635,12 +635,9 @@ class SemanticsTester {
     if (nodeData.actions != 0)
       buf.writeln('  actions: ${_actionsToSemanticsActionExpression(nodeData.actions)},');
     if (node.label != null && node.label.isNotEmpty) {
-      final String escapedLabel = node.label.replaceAll('\n', r'\n');
-      if (escapedLabel != node.label) {
-        buf.writeln("  label: r'$escapedLabel',");
-      } else {
-        buf.writeln("  label: '$escapedLabel',");
-      }
+      // Escape newlines and text directionality control characters.
+      final String escapedLabel = node.label.replaceAll('\n', r'\n').replaceAll('\u202a', '\\u202a').replaceAll('\u202c', '\\u202c');
```

Comment:
Maybe not essential for test applications but the complete list of suspect codes is:

U+202A, U+202B, U+202C, U+202D, U+202E, U+2066, U+2067, U+2068, U+2069

Reply:
Rust has a nice CVE here: https://blog.rust-lang.org/2021/11/01/cve-2021-42574.html